rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -355,7 +355,7 @@ impl PendingOperation for PendingMessage {
             if self
                 .ctx
                 .destination_mailbox
-                .process_estimate_costs(&self.message, metadata)
+                .process_estimate_costs(&self.message, metadata, true)
```

Comment:
any reason why this needs to be true?

iiuc this being true or false doesn't change the behavior we care about, which is just that an eth_estimateGas is successful. We don't care about the buffer being applied or anything here

I think if it doesn't matter, then we can keep the `process_estimate_costs` interface the same as before, and just have `process_contract_call` be the one with the interface change, and have `process_estimate_costs` always pass false as this param to `process_contract_call`. I think this would be kinda nice because it'd keep the Mailbox trait impl more self explanatory / simple